### PR TITLE
create_new_reference_messages_server_side

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

File: packages/client/src/api/wargames_api/index.ts [hunk 1]
```diff
@@ -682,36 +682,19 @@ export const postFeedback = (dbName: string, fromDetails: MessageDetailsFrom, tu
   return db.put(feedback).catch(rejectDefault)
 }
 
-const checkReference = (message: MessageCustom, db: ApiWargameDb, details: MessageDetails) => {
+const checkReference = (message: MessageCustom, db: ApiWargameDb, details: MessageDetails): Promise<MessageCustom> => {
   // eslint-disable-next-line no-async-promise-executor
   return new Promise(async (resolve): Promise<void> => {
     if (message.details.messageType !== 'Chat') {
       // default value for message counter
+
```
Comment (IanMayo) [inline packages/client/src/api/wargames_api/index.ts]: @lilitkarapetyan - I'm sure we only need to do this if `message.message.Reference` is present.

**Comment by IanMayo** [comment]
@lilitkarapetyan  - how would you feel about storing `counter` in `message.details` rather than `message.message`?

If it's in `message.message` then `json-editor` will try to render a control for it - so we'll always have to hide it via `display:none`.  We also don't have much of a schema for `message.message`.  We can add `counter?: number` to `message.details` though.

**Comment by lilitkarapetyan** [comment]
Hello Ian, I think it's a good idea message.message will only have information about the json-editor form value

**Comment by IanMayo** [comment]
> Hello Ian, I think it's a good idea message.message will only have information about the json-editor form value

I have just pushed a typescript interface change to this branch,  to introduce the `details.counter`. Are you happy to make the change to using it?

**Comment by lilitkarapetyan** [comment]
> > Hello Ian, I think it's a good idea message.message will only have information about the json-editor form value
> 
> I have just pushed a typescript interface change to this branch, to introduce the `details.counter`. Are you happy to make the change to using it?

Yes sure thanks for the change.

**Comment by IanMayo** [comment]
Nice work @lilitkarapetyan 

Could you do some testing against `yarn develop` - and check that DB-Browser shows the data updating how we'd expect.

Aah, the data in `wargame-l6nngxlk.sqlite` won't be how Serge is expecting it.  It would be useful for me to keep those messages in the game though.   Hmm, I do have code to generate messages. I could modify it to generate valid `counter` and `Reference` numbers.  If I do that, we could push the messages into `sqlite`.

Ok. Right. What are we going to do ;-)

Ok. Could you just check that things _roughly_ work correctly in DB-Browser?  The first message won't find an existing `message.detail.counter` - so it will become `Blue-1`.  If that happens, then it's ok.  But, if you then modify that `message.detail.counter` to have a value of `20` - then new messages should get handled properly - becoming `Blue-21`, and so on.

**Comment by lilitkarapetyan** [comment]
> Nice work @lilitkarapetyan
> 
> Could you do some testing against `yarn develop` - and check that DB-Browser shows the data updating how we'd expect.
> 
> Aah, the data in `wargame-l6nngxlk.sqlite` won't be how Serge is expecting it. It would be useful for me to keep those messages in the game though. Hmm, I do have code to generate messages. I could modify it to generate valid `counter` and `Reference` numbers. If I do that, we could push the messages into `sqlite`.
> 
> Ok. Right. What are we going to do ;-)
> 
> Ok. Could you just check that things _roughly_ work correctly in DB-Browser? The first message won't find an existing `message.detail.counter` - so it will become `Blue-1`. If that happens, then it's ok. But, if you then modify that `message.detail.counter` to have a value of `20` - then new messages should get handled properly - becoming `Blue-21`, and so on.
Thanks, I will try with value 20 now

**Comment by lilitkarapetyan** [comment]
I tested it, it works without problems

**Comment by lilitkarapetyan** [comment]
You can test it `P9 Test` in the `BLUE RFI` channel of the game.

**Review by IanMayo — APPROVED** [review]
Working fine

**Pull request merged**